spec/generators/statesman/migration_generator_spec.rb

```diff
@@ -39,4 +39,24 @@
         to contain("name: \"index_bacon_transitions_parent_most_recent\"")
     end
   end
+
+  describe "properly inherit class" do
+    subject do
+      file(
+        "db/migrate/#{migration_number}_add_statesman_to_bacon_transitions.rb"
+      )
+    end
+
+    before do
+      allow(Time).to receive(:now).and_return(mock_time)
+      run_generator %w(Yummy::Bacon Yummy::BaconTransition)
+    end
+
+    if ::ActiveRecord.respond_to?(:gem_version) &&
+       ::ActiveRecord.gem_version >= Gem::Version.new('5.0.0')
+      it { is_expected.to contain(/ActiveRecord::Migration\[\d+\.\d+\]/) }
+    else
+      it { is_expected.not_to contain(/ActiveRecord::Migration\[\d+\.\d+\]/) }
+    end
+  end
```

Comment:
So I'd prefer to mock out `ActiveRecord.gem_version` to return `5.0.0` and expect that the output matches a static fixture, rather than testing it dynamically like this.